ts/features/newWallet/screens/WalletHomeScreen.tsx

```diff
@@ -44,12 +46,23 @@ const WalletHomeScreen = ({ route }: Props) => {
     React.useCallback(() => {
       trackOpenWalletScreen();
       dispatch(walletToggleLoadingState(true));
-      dispatch(getPaymentsWalletUserMethods.request());
-      dispatch(idPayWalletGet.request());
-      dispatch(cgnDetails.request());
     }, [dispatch])
   );
 
+  useOnFirstRender(() => {
+    fetchWalletSectionData();
+  });
+
+  useOnWalletUserSessionRefresh(() => {
+    fetchWalletSectionData();
+  });
+
+  const fetchWalletSectionData = () => {
+    dispatch(getPaymentsWalletUserMethods.request());
+    dispatch(idPayWalletGet.request());
+    dispatch(cgnDetails.request());
+  };
```

Comment:
The loading state should be triggered before we fetch something, so it should be moved here.

```suggestion
  const fetchWalletSectionData = () => {
    dispatch(walletToggleLoadingState(true));
    dispatch(getPaymentsWalletUserMethods.request());
    dispatch(idPayWalletGet.request());
    dispatch(cgnDetails.request());
  };
```